### DISCUSSION TITLE
Android Wear App

### DISCUSSION THREAD

**Discussion by Querschlag** [iSoron/uhabits#82] [Feature Requests]
This is kind of a stretch goal, but perfect for this type of app.
Setting a habit check from your wrist without having to pull out your phone would be a real improvement during daily usage.

For the last 2 years I used Tickmate for my habit tracking. Last year I developed an Android Wear App for this project (currently not merged yet).
I would be happy to contribute with my experience.

Tickmate Wear: https://github.com/lordi/tickmate/issues/57

**Comment by iSoron** [comment]
Yes, the app currently has some support for Android wear via notifications, but it would be really nice to have a proper app running on the watch itself. If you could contribute with some code, that would be very welcome, of course.

I will download and try your Tickmate fork on my watch later today.

**Comment by matthewfcarlson** [comment]
I'd be happy to work on this

**Comment by NellyWhads** [comment]
Did this end up taking off?

**Comment by matthewfcarlson** [comment]
I started but ran into some issues and got caught up in other things. I don't know if I ever had something worthy of pushing up

**Comment by iSoron** [comment]
I still consider this an interesting feature to have. Prototypes are welcome.

**Comment by NellyWhads** [comment]
Thinking more about how to implement this, it seems we run into similar issues as if we were implementing multi-device-sync.

- How to handle "completion" on one device and propagate to the other seamlessly
- How to handle offline changes.

The first could be resolved with a simple master-slave relationship. Phone is always master, Wear OS is always slave. Queries on Wear OS app-launch can update the interface explicitly.

For the second, could we only enable changes if the phone is connected. If so, the resolution to the first point works everything out. If anything, this could be a first pass solution and a good way to prototype multi-device-sync type feature while ignoring cloud database privacy-related blockers.

Only "watched" (pun completely intended) habits can be made available on the Wear OS device allowing for a clean watch interface without massive scrolling lists on tiny screens.

**Comment by iSoron** [comment]
For a first version, I wouldn't mind if the watch app required a constant connection to the phone. It could not even store any data locally; just request it from the phone and display it. When we have device sync, the watch app could become another full client.

**Comment by f5b7** [comment]
NB: There is a more recent discussion about this, having more votes:
https://github.com/iSoron/uhabits/discussions/1507
